From 31a09167fdc4b53e618d3ac2639767bf5eb431f8 Mon Sep 17 00:00:00 2001
From: LevyHsu <admin@levyhsu.com>
Date: Fri, 16 Jul 2021 10:29:25 +0800
Subject: [PATCH 61/61] Remove unsupported bext.w in gcc

Followiing code was removed to prevent bextw being generated:


(define_insn "*bextw"
  [(set (match_operand:DI 0 "register_operand" "=r")
	(and:DI
	 (subreg:DI
	  (lshiftrt:SI (match_operand:SI 1 "register_operand" "r")
		       (match_operand:QI 2 "register_operand" "r")) 0)
	 (const_int 1)))]
  "TARGET_64BIT && TARGET_ZBS"
  "bextw\t%0,%1,%2"
  [(set_attr "type" "bitmanip")])
---
 gcc/config/riscv/bitmanip.md | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/gcc/config/riscv/bitmanip.md b/gcc/config/riscv/bitmanip.md
index 3026883fc0e..930549a90fc 100644
--- a/gcc/config/riscv/bitmanip.md
+++ b/gcc/config/riscv/bitmanip.md
@@ -206,17 +206,6 @@
   "bexti\t%0,%1,%2"
   [(set_attr "type" "bitmanip")])
 
-(define_insn "*bextw"
-  [(set (match_operand:DI 0 "register_operand" "=r")
-	(and:DI
-	 (subreg:DI
-	  (lshiftrt:SI (match_operand:SI 1 "register_operand" "r")
-		       (match_operand:QI 2 "register_operand" "r")) 0)
-	 (const_int 1)))]
-  "TARGET_64BIT && TARGET_ZBS"
-  "bextw\t%0,%1,%2"
-  [(set_attr "type" "bitmanip")])
-
 ;;; ??? s[lr]o*
 
 (define_insn "rotrsi3"
-- 
2.33.0

